In [3]:
#https://courses.thinkful.com/trial-data-001/project/3.1.5
#setting up dataframe
import pandas as pd
import numpy as np
import statistics
import matplotlib.pyplot as plt

bbdf= pd.DataFrame()
bbdf['age']=[14,12,11,10,8,6,8]
bbdf['names']=['Greg','Marcia','Peter','Jan','Bobby','Cindy','Cousin Oliver']
print(bbdf)

   age   names
0   14    Greg
1   12  Marcia
2   11   Peter
3   10     Jan
4    8   Bobby
5    6   Cindy
6    8  Oliver


In [12]:
bbdf.describe()

age
count   7.000000
mean    9.857143
std     2.734262
min     6.000000
25%     8.000000
50%    10.000000
75%    11.500000
max    14.000000

In [26]:
#central tendency
print(statistics.median(bbdf['age']))
print(np.mean(bbdf['age']))
print(statistics.mode(bbdf['age']))


10
9.857142857142858
8


In [24]:
#variance

print(bbdf['age'].var())
print(np.std(bbdf['age'], ddof=1))
print(np.std(bbdf['age'], ddof=1)/np.sqrt(7))

7.476190476190475
2.734262327610589
1.0334540197243192


If I had to choose any one marker of central tendency I would pick the mean and median. Given that the median and mean are close, we can tell quite a bit the data from the combination of these two outputs. The mode, meanwhile, only tells us there are two 8 year olds. It doesn't give us any information about the point towards which the dataset centers around. In this case, the mean and median do a better job of this.

With regards the measures of variance. I would pick the the standard deviation. It gives us a good  The Brady Bunch are a family, who from my understanding have 1 mother (except for Cousin Oliver, who perhaps should be denoted a string variable that donates him as a cousin rather than a sibling). In either case, the standard error isn't really all that meaningful given that, assuming that the mother isn't hyper fertile, she can only have 1 child every 11 months. Due to this, the standard error should tend towards 1 as the mother has more children. This doesn't tell us anything we do not already know. The standard deviation is relatively reflective of the data in this case. All individuals except for Greg and Cindy fall within 1 standard deviation from the mean.

In [34]:
bbdf.iloc[5]=[7, 'Cindy']
bbdf.iloc[6]=[1, 'Jessica']
print(bbdf)

   age    names
0   14     Greg
1   12   Marcia
2   11    Peter
3   10      Jan
4    8    Bobby
5    7    Cindy
6    1  Jessica


In [36]:
#let's recheck central tendency and variance

#central tendency
print(statistics.median(bbdf['age']))
print(np.mean(bbdf['age']))
#print(statistics.mode(bbdf['age'])) - there is no mode for this dataset anymore

#variance

print(bbdf['age'].var())
print(np.std(bbdf['age'], ddof=1))
print(np.std(bbdf['age'], ddof=1)/np.sqrt(7))

10
9.0
18.0
4.242640687119285
1.6035674514745462


Median is still 10, mean has dropped to 7, there no longer is a mode for this dataset and our measures of variance has shot up quite a bit. I still think the mean and standard deviation is the most descriptive combination.

A quick analysis of the ratio of our standard deviation (4.24) to our mean (9.0). Tells you this dataset now has quite high variance, this is meaningful data and should not be hidden  from plain view. I prefer to know what i'm working with up front. Beyond that, most descriptive statistics calculations use the mean not the median. This is because the mean is more mathematically meaningful.

In [40]:
bb_50thsurvey_df= pd.DataFrame()
bb_50thsurvey_df['survey_data']=[20, 23, 17, 5]
bb_50thsurvey_df['survey_sources']=['TV Guide', 'Entertainment Weekly', ' Pop Culture Today', ' SciPhi Phanatic']
print(bb_50thsurvey_df)

   survey_data        survey_sources
0           20              TV Guide
1           23  Entertainment Weekly
2           17     Pop Culture Today
3            5       SciPhi Phanatic


In [47]:
print(np.mean(bb_50thsurvey_df['survey_data'][0:3]))

print(np.std(bb_50thsurvey_df[['survey_data'][0:3]], ddof=1)/np.sqrt(3))

20.0
survey_data    4.555217
dtype: float64


I calculated the mean and standard error for the TV Guide, Entertainment Weekly and Pop Culture Today data. I omitted the SciPhi Phanatic data as it is a clear outlier and beyond that, from a practical stand point, estimates suggest that 1/5 Americans are sci-fi fans and that 1/5 are predominantly men [1]. Accordingly, I don't think survey's sampling is as representative of the general population as the others.

Our data suggests that between 15.5% and 24.5% are fans of the Brady Bunch. This data of course, has it's assumptions, most notably that the preferences of American magazine readers reflect all Americans.
 

https://www.sfwa.org/2014/01/reads-science-fiction/

